components/ai_chat/core/browser/engine/oai_parsing.cc

```diff
@@ -0,0 +1,139 @@
+// Copyright (c) 2025 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#include "brave/components/ai_chat/core/browser/engine/oai_parsing.h"
+
+#include <string>
+#include <utility>
+
+#include "brave/components/ai_chat/core/browser/engine/engine_consumer.h"
+#include "brave/components/ai_chat/core/browser/tools/tool.h"
+
+namespace ai_chat {
+
+std::vector<mojom::ToolUseEventPtr> ToolUseEventFromToolCallsResponse(
+    const base::Value::List* tool_calls_api_response) {
+  // https://platform.openai.com/docs/api-reference/chat/create#chat-create-tools
+  // https://platform.openai.com/docs/api-reference/chat/object
+  // choices -> message -> tool_calls
+  std::vector<mojom::ToolUseEventPtr> tool_use_events;
+  for (auto& tool_call_raw : *tool_calls_api_response) {
+    if (!tool_call_raw.is_dict()) {
+      DLOG(ERROR) << "Tool call is not a dictionary.";
+      continue;
+    }
+    const auto& tool_call = tool_call_raw.GetDict();
+
+    // Most APIs that have partial chunk responses seem to initially always have
+    // the tool name and id, and only chunk the arguments json. So, for now, we
+    // can continue to have validation here and ignore calls that don't have
+    // required fields.
+
+    const base::Value::Dict* function = tool_call.FindDict("function");
+    if (!function) {
+      DLOG(ERROR) << "No function info found in tool call.";
+      continue;
+    }
+    // Tool call results can be partial and should be added to the previous
+    // event by the event handler.
+    const std::string* id = tool_call.FindString("id");
+    if (!id) {
+      DLOG(ERROR) << "No id found in tool call.";
+      continue;
+    }
+
+    const std::string* name = function->FindString("name");
+    if (!name) {
+      DLOG(ERROR) << "No name found in tool call.";
+      continue;
+    }
+
+    mojom::ToolUseEventPtr tool_use_event =
+        mojom::ToolUseEvent::New(*name, *id, "", std::nullopt, std::nullopt);
+
+    // Arguments are provided as a JSON string. We parse it here so that the
+    // rest of the code doesn't have to parse it potentially multiple times
+    // (performance), and we can add some protections.
```

Comment:
note: we aren't actually parsing this here - maybe we should remove this comment?